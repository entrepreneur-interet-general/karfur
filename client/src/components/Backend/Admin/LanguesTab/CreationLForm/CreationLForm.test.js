import React from 'react';
import { expect } from 'chai';
import { shallow, mount } from 'enzyme';
import CreationLForm from './CreationLForm';

const langue= {};

const langues=[{
  "_id" : "5c8b89e8d492774888f1acaf",
  "langueFr" : "Arabe",
  "langueLoc" : "العربية",
  "langueCode" : "sa",
  "langueIsDialect" : false,
  "langueBackupId" : "undefined",
  "status" : "Active",
  "i18nCode" : "ar",
  "created_at" : "2019-03-15T11:18:00.538Z",
  "updatedAt" : "2019-03-23T23:09:05.242Z",
  "__v" : 15,
  "participants" : [ 
    "5c8f5affa56cba6f5b82db60", 
    "5c8f5affa56cba6f5b82db62", 
    "5c8f5affa56cba6f5b82db63", 
    "5c95099f5de3940eca6b7354", 
    "5c950b3b5de3940eca6b7364", 
    "5c96212e5de3940eca6b73e7", 
    "5c9623945de3940eca6b741e", 
    "5c96bbd65de3940eca6b7452", 
    "5c96bc7e5de3940eca6b745f"
  ]
},
{
  "_id" : "5c8b918d838c594a9884f561",
  "langueFr" : "Anglais",
  "langueLoc" : "English",
  "langueCode" : "gb",
  "i18nCode" : "en",
  "langueIsDialect" : false,
  "langueBackupId" : "undefined",
  "status" : "Active",
  "created_at" : "2019-03-15T11:50:37.877Z",
  "updatedAt" : "2019-03-23T12:16:53.843Z",
  "__v" : 11,
  "participants" : [ 
      "5c8f5affa56cba6f5b82db60", 
      "5c94af8b25aa640c19f35c00", 
      "5c95099f5de3940eca6b7354", 
      "5c950b3b5de3940eca6b7364", 
      "5c96212e5de3940eca6b73e7", 
      "5c9623945de3940eca6b741e"
  ]
}]

const handleChange = () => {};

describe('<CreationLForm />', () => {

  it('mounts without crashing', () => {
    const wrapper = shallow(<CreationLForm langue={langue} langues={langues} handleChange={handleChange}/>);
    wrapper.unmount()
  });

  it('should match its reference snapshot', () => {
    const wrapper = shallow(<CreationLForm langue={langue} langues={langues} handleChange={handleChange} />)
  
    expect(wrapper).to.matchSnapshot()
  })

  // it('collapse without crashing', () => {
  //   const wrapper = mount(<CreationLForm langue={langue} langues={langues} />);
  //   let selectInput = wrapper.find('#langueBackupId');
  //   console.log(selectInput)
  //   selectInput.simulate('click');
  //   // expect(wrapper.state().collapse).toEqual(false);
  //   // collapse.simulate('click');
  //   // expect(wrapper.state().collapse).toEqual(true);
  //   // wrapper.unmount()
  // });

  it('should edit input on key press', () => {
    const wrapper = shallow(<CreationLForm langue={langue} langues={langues} handleChange={handleChange} />)
  });

  it ('accepts langue props', () => {
    const wrapper = mount(<CreationLForm langue={langues[0]} langues={langues} handleChange={handleChange} />);
    expect(wrapper.props().langue).equal(langues[0])
  })
})