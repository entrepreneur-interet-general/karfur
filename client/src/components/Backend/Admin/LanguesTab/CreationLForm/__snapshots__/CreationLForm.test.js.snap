// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreationLForm /> should match its reference snapshot 1`] = `
<Card
  tag="div"
>
  <CardHeader
    className="h1"
    tag="div"
  >
    <strong>
      Ajouter une nouvelle langue
    </strong>
  </CardHeader>
  <CardBody
    tag="div"
  >
    <Form
      action=""
      className="form-horizontal"
      encType="multipart/form-data"
      method="post"
      tag="form"
    >
      <FormGroup
        row={true}
        tag="div"
      >
        <Col
          md="3"
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <Label
            htmlFor="langueFr"
            tag="label"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            Nom de la langue en français
          </Label>
        </Col>
        <Col
          md="9"
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
          xs="12"
        >
          <Input
            id="langueFr"
            name="langue"
            onChange={[Function]}
            placeholder="Anglais"
            type="text"
          />
          <FormText
            color="muted"
            tag="small"
          >
            Par exemple : Anglais
          </FormText>
        </Col>
      </FormGroup>
      <FormGroup
        row={true}
        tag="div"
      >
        <Col
          md="3"
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <Label
            htmlFor="langueLoc"
            tag="label"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            Nom de la langue en local
          </Label>
        </Col>
        <Col
          md="9"
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
          xs="12"
        >
          <Input
            id="langueLoc"
            name="langue"
            onChange={[Function]}
            placeholder="English"
            type="text"
          />
          <FormText
            color="muted"
            tag="small"
          >
            Par exemple : English
          </FormText>
        </Col>
      </FormGroup>
      <FormGroup
        row={true}
        tag="div"
      >
        <Col
          md="3"
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <Label
            htmlFor="langueCode"
            tag="label"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            Code du drapeau
          </Label>
        </Col>
        <Col
          md="9"
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
          xs="12"
        >
          <Input
            id="langueCode"
            name="langue"
            onChange={[Function]}
            placeholder="gb"
            type="text"
          />
          <FormText
            color="muted"
            tag="small"
          >
            Le drapeau résultant : 
            <i
              className="flag-icon flag-icon-undefined"
            />
          </FormText>
        </Col>
      </FormGroup>
      <FormGroup
        row={true}
        tag="div"
      >
        <Col
          md="3"
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <Label
            htmlFor="i18nCode"
            tag="label"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            Code i18n
          </Label>
        </Col>
        <Col
          md="9"
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
          xs="12"
        >
          <Input
            id="i18nCode"
            name="langue"
            onChange={[Function]}
            placeholder="en"
            type="text"
          />
          <FormText
            color="muted"
            tag="small"
          >
            Au format 'en' ou 'en-GB'. Voir 
            <a
              href="http://www.i18nguy.com/unicode/language-identifiers.html"
              rel="noopener noreferrer"
              target="_blank"
            >
              ce site
            </a>
             pour le tableau de correspondance.
          </FormText>
        </Col>
      </FormGroup>
      <FormGroup
        row={true}
        tag="div"
      >
        <Col
          md="3"
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <Label
            tag="label"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            La langue est-elle un dialecte d'une autre langue ?
          </Label>
        </Col>
        <Col
          md="9"
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <FormGroup
            check={true}
            className="radio"
            inline={true}
            tag="div"
          >
            <Input
              checked={false}
              className="form-check-input"
              id="langueIsDialect"
              name="langue"
              onChange={[Function]}
              type="radio"
              value={true}
            />
            <Label
              check={true}
              className="form-check-label"
              htmlFor="langueIsDialect"
              tag="label"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
            >
              Oui
            </Label>
          </FormGroup>
          <FormGroup
            check={true}
            className="radio"
            inline={true}
            tag="div"
          >
            <Input
              checked={false}
              className="form-check-input"
              id="langueIsDialectB|><"
              name="langue"
              onChange={[Function]}
              type="radio"
              value={false}
            />
            <Label
              check={true}
              className="form-check-label"
              htmlFor="langueIsDialectB|><"
              tag="label"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
            >
              Non
            </Label>
          </FormGroup>
        </Col>
      </FormGroup>
      <FormGroup
        row={true}
        tag="div"
      >
        <Col
          md="3"
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <Label
            htmlFor="select"
            tag="label"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            Statut sur notre site
          </Label>
        </Col>
        <Col
          md="9"
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
          xs="12"
        >
          <Input
            id="statusB|><"
            name="langue"
            onChange={[Function]}
            type="select"
          >
            <option
              key="Active"
              value="Active"
            >
              Active
            </option>
            <option
              key="En attente"
              value="En attente"
            >
              En attente
            </option>
            <option
              key="Inactive"
              value="Inactive"
            >
              Inactive
            </option>
            <option
              key="Annulée"
              value="Annulée"
            >
              Annulée
            </option>
          </Input>
        </Col>
      </FormGroup>
    </Form>
  </CardBody>
  <CardFooter
    tag="div"
  >
    <Row
      tag="div"
    >
      <Col
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <Button
          block={true}
          color="success"
          onClick={[Function]}
          size="lg"
          tag="button"
        >
          Valider
        </Button>
      </Col>
      <Col
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <Button
          block={true}
          color="danger"
          onClick={[Function]}
          size="lg"
          tag="button"
        >
          Annuler
        </Button>
      </Col>
    </Row>
  </CardFooter>
</Card>
`;
